src/elements/content-sidebar/activity-feed/task-new/Task.js

```diff
@@ -34,157 +37,158 @@ type Props = {
     onEdit?: Function,
     translatedTaggedMessage?: string,
     translations?: Translations,
-};
+|};
 
-const MAX_AVATARS = 3;
+type AssigneeProps = {|
+    assignees: TaskAssigneeCollection,
+    getAvatarUrl: GetAvatarUrlCallback,
+    maxAvatars?: number,
+|};
 
-// eslint-disable-next-line react/prefer-stateless-function
-class Task extends React.Component<Props> {
-    getMessageForTask(isCurrentUser: boolean, taskType: TaskType): Object {
-        if (isCurrentUser) {
-            if (taskType === TASK_TYPE_APPROVAL) {
-                return messages.tasksFeedHeadlineApprovalCurrentUser;
-            }
-            return messages.tasksFeedHeadlineGeneralCurrentUser;
-        }
+const Assignees = React.memo<AssigneeProps>(({ maxAvatars = MAX_AVATARS, assignees, getAvatarUrl }: AssigneeProps) => {
```

Comment:
Should this be split out into its own file?

Reply:
Maybe later. There's performance benefits to creating more components, but maintenance overhead in splitting up the files.